### PR TITLE
Load phones list api

### PR DESCRIPTION
**Pull request by roma-shafihullin**

### CODE REVIEW

File: fe/src/components/Pagination/PaginationBase.tsx [hunk 1]
```diff
@@ -57,8 +68,8 @@ export const PaginationBase: React.FC = () => {
       </div>
       <div className="container">
         <ul className="grid">
-          {visibleItems.map((item) => (
-            <li key={item}>{<ProductCard />}</li>
+          {products.map((item) => (
+            <li key={item}>{<ProductCard phone={item} />}</li>
```
Comment (VolodymyrChuyko) [inline fe/src/components/Pagination/PaginationBase.tsx:72]:
```suggestion
            <li key={item.id}><ProductCard phone={item} /></li>
```

File: fe/src/api/phones.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import axios from 'axios';
+
+axios.defaults.baseURL = 'https://project-nice-gadgets.herokuapp.com/';
+
+export function getPhones() {
+  return axios.get('/products');
+}
```
Comment (VolodymyrChuyko) [inline fe/src/api/phones.ts:7]: You also need to implement a function that adds a query to the request url

File: fe/src/components/Pagination/PaginationBase.tsx [hunk 3]
```diff
@@ -18,6 +20,15 @@ const items = getNumbers(1, 15).map((n) => `Item ${n}`);
...
   const [currentPage, setCurrentPage] = useState(1);
   const [itemsPerPage, setItemsPerPage] = useState(4);
+  const [products, setProducts] = useState([]);
+
+/* eslint-disable */
+  useEffect(() => {
+    getPhones()
+      .then((res) => res.data)
+      .then(setProducts)
+      .catch(error => console.log(error));
```
Comment (VolodymyrChuyko) [inline fe/src/components/Pagination/PaginationBase.tsx:30]: you should request data using query params

**Review by VolodymyrChuyko — CHANGES_REQUESTED** [review]
image
Also, I've changed the response array. Now item with a 0 index is the max number of products. You should implement .slice(1) to the response array before mapping the products

File: fe/src/api/phones.ts (hunk 2 above)
Comment (roma-shafihullin) [reply fe/src/api/phones.ts]: > You also need to implement a function that adds a query to the request url

IMAGE 2022-11-04 14:15:53

**Review by VolodymyrChuyko — APPROVED** [review]
Let's merge it!

**Pull request merged**